src/class-post-type.php

```diff
@@ -0,0 +1,85 @@
+			'name'                  => __( $plural, $translation_domain ),
+			'singular_name'         => __( $singular, $translation_domain ),
+			'menu_name'             => __( $plural, $translation_domain ),
+			'name_admin_bar'        => __( $singular, $translation_domain ),
+			'all_items'             => __( 'All ' . $plural_lowercase, $translation_domain ),
+			'add_new_item'          => __( 'Add a ' . $singular_lowercase, $translation_domain ),
+			'edit_item'             => __( 'Edit ' . $singular_lowercase, $translation_domain ),
+			'new_item'              => __( 'New ' . $singular_lowercase, $translation_domain ),
+			'view_item'             => __( 'See ' . $singular_lowercase, $translation_domain ),
+			'view_items'            => __( 'See ' . $plural_lowercase, $translation_domain ),
+			'archives'              => __( $plural_lowercase . ' archives', $translation_domain ),
+			'attributes'            => __( $plural_lowercase . ' attributs', $translation_domain ),
+			'filter_items_list'     => __( 'Filter ' . $plural_lowercase . ' list', $translation_domain ),
+			'items_list'            => __( $plural_lowercase . ' list', $translation_domain ),
+			'items_list_navigation' => __( $plural_lowercase . ' list navigation', $translation_domain ),
+			'search_items'          => __( 'Search ' . $plural_lowercase, $translation_domain ),
+			'not_found'             => __( 'No ' . $singular_lowercase, $translation_domain ),
+			'not_found_in_trash'    => __( 'No ' . $singular_lowercase . ' in trash', $translation_domain ),
+		];
+		// phpcs:enable
+
+		return $labels;
+	}
+
+	/**
+	 * Create a custom post type
+	 *
+	 * @param  string       $singular           Singular name.
+	 * @param  string       $plural             Plural name.
+	 * @param  string|null  $translation_domain Translation domain name.
+	 * @param  array        $args               Custom arguments.
+	 * @param  bool|boolean $public             Is post type public?.
+	 * @param  array        $supports           Supported features.
+	 * @return array                            Custom post type
+	 */
+	public static function create( string $singular, string $plural, string $translation_domain = null, array $args = array(), bool $public = true, array $supports = array() ) {
+		$singular_lowercase = strtolower( $singular );
+
+		$labels = self::generate_labels( $singular, $plural, $translation_domain );
+
+		// phpcs:disable WordPress.WP.I18n
+		$default_args = array(
+			'label'              => __( $singular, $translation_domain ),
+			'labels'             => $labels,
+			'public'             => true,
+			'publicly_queryable' => $public,
+			'capability_type'    => 'post',
+			'supports'           => array(
+				'title',
+				'custom-fields',
+			),
+			'has_archive'        => false,
+			'menu_icon'          => 'dashicons-format-aside',
+		);
+		// phpcs:enable
+
+		$args = array_merge( $default_args, $args );
+
+		$post_type = register_post_type( $singular_lowercase, $args );
```

Comment:
I'm not sure we should use the labels to create the machine name for a cpt. 
We want our machine name to always be in english, but we might want our labels to be in French for the customer experience.

Reply:
If the label should be in french, the machine name must be in french too. So, the current behaviour allow this.

It is a bad practice to have the machine name in english and labels in french, we should keep consistency between labels and machine name.

Reply:
I think i disagree with you, labels are for humans and machine names are for machines (and devs).
For exemple the default WP post_type have a machine name of `post` and labels are in the langage of your WP installation. All the machine langage is in english in WP (terms, taxonomy, post ..)

What's your opinion @studiometa/wordpress ?

Reply:
We can manage this behaviour with i18n plugin. Like Wordpress does. Register the CPT machine name in english, translate it with i18n plugin.

Reply:
I just had a case that (still) makes me think we should not use labels for machine names. 
If you want to create a CPT like `Customer Story`, you end up with `customerstory` for the machine name, you cannot have `customer-story`